enhancements/kube-apiserver/unsupported-file-sync.md

```diff
@@ -0,0 +1,153 @@
+  - tkashem
+  - deads2k
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - tkashem
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: 
+  - AUTH-387
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Unsupported file synchronization into kube-apiserver static pods
+
+## Summary
+
+Since kube-apiserver pods are running as static, it's hard to test upcoming features
+that might require configuration via files. This enhancement describes how to synchronize
+files to kube-apiserver pods in an unsupported mode that matches the testing use-case.
+
+## Motivation
+
+Simplify testing features that might require additional files to be installed into
+kube-apiserver pods.
+
+### User Stories
+
+* As a developer, I want to be able to create configuration/trust files in kube-apiserver
+
+### Goals
+
+- allow mounting unsupported content as files in the kube-apiserver static pods
+
+### Non-Goals
+
+- allow creating files anywhere on a node
+
+## Proposal
+
+The `openshift-kube-apiserver-operator` namespace gets a new revisioned
+secret `unsupported-kube-apiserver-content` that is going to be used to
+install content inside of the `/etc/kubernetes/unsupported/kube-apiserver` directory.
+This secret will be populated by an observer inside of the cluster-kube-apiserver-operator.
+The observer watches the `unsupported-kube-apiserver-content` secret in
+the `openshift-config` namespace and synchronizes its content to the
+secret inside of the `openshift-kube-apiserver-namespace`.
+
+### Workflow Description
+
+The user adds content to the `data` and `stringData` maps in the
+`openshift-config/unsupported-kube-apiserver-content` secret and
+waits for a rollout of a new kube-apiserver revision that contains their changes.
+
+### API Extensions
+
+No API extensions.
+
+### Implementation Details
+
+#### New kube-apiserver operator observer
+
+A new observer is added to the cluster-kube-apiserver-operator. This observer
+watches the `data` and `stringData` fields of the `openshift-config/unsupported-kube-apiserver-content`
+secrets and synchronizes them to the `openshift-kube-apiserver-operator/unsupported-kube-apiserver-content`
+secret.
+
+If the `data` or `stringData` fields of the source secret are non-empty, the observer
+sets the `UnsupportedFilesUpgradeable: False` condition in the `kube-apiserver`
```

Comment:
Wouldn't it be best to split `UnsupportedConfigFilesPresent: True` and `Upgradable: False` - iiuc CVO still looks for Upgradable only

Reply:
No. The `Upgradeable` clusteroperator is a condition that is aggregated from all `*Upgradeable` conditions set in the `kubeapiservers.operator.openshift.io/cluster` objects.

I should perhaps clarify that by `kube-apiserver` clusteroperator I actually mean the `operator` API group object.

Reply:
See https://github.com/openshift/library-go/blob/d7e7beca5baea998564ed16bd3d7e7dffbc7c01f/pkg/operator/status/status_controller.go#L123 for how the condition aggregation works.

Reply:
Ah, I was not aware. LGTM then